src/compiler/crystal/tools/doc/type.cr

```diff
@@ -618,49 +616,52 @@ class Crystal::Doc::Type
   end
 
   def type_to_html(type : Crystal::GenericInstanceType, io, text = nil, links = true)
+    not_instantiated = type.type_vars.all? { |(name, type_var)| type_var.is_a?(Var) && type_var.type.is_a?(TypeParameter) }
     generic_type = @generator.type(type.generic_type)
-    if generic_type.must_be_included?
-      if links
-        io << %(<a href=")
-        io << generic_type.path_from(self)
-        io << %(">)
-      end
-      if text
+    must_be_included = generic_type.must_be_included?
+    if must_be_included && links
+      io << %(<a href=")
+      io << generic_type.path_from(self)
+      io << %(">)
+    end
+
+    if text
         io << text
-      else
-        generic_type.full_name_without_type_vars(io)
-      end
-      if links
-        io << "</a>"
-      end
     else
-      if text
-        io << text
-      else
-        generic_type.full_name_without_type_vars(io)
+      generic_type.full_name_without_type_vars(io)
+      if not_instantiated
+        io << '('
+        type.type_vars.values.join(", ") { |type_var| type_var.as(Var).type.to_s io }
```

Comment:
Could that be shortened to `type.type_vars.values.join ", ", &.as(Var).type.to_s(io)`?